### PR TITLE
fix: handle async placeholders in normalizeScopedSlot 

### LINKED ISSUE #10391: Nested async components passed as slots are not rendered in SSR
### Version
2.6.10

### Reproduction link
[https://codesandbox.io/s/vue-ssr-nested-async-components-repro-script-ozze7](https://codesandbox.io/s/vue-ssr-nested-async-components-repro-script-ozze7)

### Steps to reproduce
When the codesandbox starts up, it should run the ```npm start``` script. If it doesn't, open a bash terminal and run ```npm start``` to see the HTML generated from SSR. You can also take a look at the components in the ```src``` folder to see how they are configured.

### What is expected?
Expected the HTML for the nested async component (called "SomeComponent") to be rendered during SSR:

```html
<div data-server-rendered="true">
  <div class="Grid">
    <div class="Grid__left">Left
      <div class="SomeComponent">Somecomponent with value: first</div>
    </div>
    <div class="Grid__right">Right
      <div class="SomeComponent">Somecomponent with value: second</div>
    </div>
  </div>
</div>
```

### What is actually happening?
The HTML for the nested async component (called "SomeComponent") is not rendered during SSR. Only the HTML for the outer async component (called "Grid" ) is rendered:

```html
<div data-server-rendered="true">
  <div class="Grid">
    <div class="Grid__left">Left
    </div>
    <div class="Grid__right">Right
    </div>
  </div>
</div>
```

---
I have an App that is composed of many async components. Some of these components have named slots. I was trying to build a render function that will accept a config object to render these components in various setups (for example, an async component set to a named slot of another async component). I noticed that if I nest async components within another, the HTML for the async component does not render during SSR. The codesandbox is simplified scenario of this

### PR DESCRIPTION
**Pull request by pi0**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [ ] New/updated tests are included

**Context:**

- Reproduction: https://replit.com/@pi0/vuejsvue10391
- resolves #10391, resolves nuxt/nuxt.js#8904. resolves nuxt/components#164

### CODE REVIEW

File: src/core/vdom/helpers/normalize-scoped-slots.js [hunk 1]
```diff
@@ -60,9 +61,10 @@ function normalizeScopedSlot(normalSlots, key, fn) {
     res = res && typeof res === 'object' && !Array.isArray(res)
       ? [res] // single vnode
       : normalizeChildren(res)
+    let vnode: VNode = res && res[0]
```
Comment (posva) [inline src/core/vdom/helpers/normalize-scoped-slots.js:64]: shouldn't we keep the length check of 1?
```suggestion
    let vnode: VNode = res && res.length === 1 && res[0]
```

**Review by posva — COMMENTED** [review]
Thanks for this!

File: src/core/vdom/helpers/normalize-scoped-slots.js [hunk 2]
```diff
@@ -60,9 +61,10 @@ function normalizeScopedSlot(normalSlots, key, fn) {
     res = res && typeof res === 'object' && !Array.isArray(res)
       ? [res] // single vnode
       : normalizeChildren(res)
+    let vnode: VNode = res && res.length === 1 && res[0]
```
Comment (posva) [inline src/core/vdom/helpers/normalize-scoped-slots.js:64]:
```suggestion
    let vnode: VNode = res && res[0]
```

I was wrong I guess

**Comment by patrioticcow** [comment]
Can we merge this in? I need this

File: src/core/vdom/helpers/normalize-scoped-slots.js (hunk 1 above)
Comment (caugner) [reply src/core/vdom/helpers/normalize-scoped-slots.js:64]: This conversation was marked as resolved, but the suggested change didn't make it into the MR, and possibly caused #12102.

**Pull request merged**